numpy/lib/tests/test_function_base.py

```diff
@@ -2307,6 +2307,20 @@ def test_writeback(self):
         assert_equal(x[0, :], 0)
         assert_equal(x[1, :], X)
 
+    def test_higher_dimensions(self):
+        a, b, c = np.meshgrid([0], [1, 1], [2, 2])
```

Comment:
This would be a better test if something like `np.meshgrid([0], [1, 2], [3, 4, 5])` was used.

Reply:
Could also break this into two functions: `test_nd_indexing` and `test_nd_shape`. I think the testing suite here could be improved further but it would take some thinking about how to generate the expected results without using meshgrid itself. Might be able to do that using `itertools.product`.

Reply:
> This would be a better test if something like np.meshgrid([0], [1, 2], [3, 4, 5]) was used.

Done.

> Could also break this into two functions: test_nd_indexing and test_nd_shape.

Done.

> I think the testing suite here could be improved further but it would take some thinking about how to generate the expected results without using meshgrid itself.

Do you want to do this in the same PR?

One alternate implementation of this function resides in TensorFlow: https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/python/ops/array_ops.py#L3476-L3553

One thing to be aware of is that since TF's implementation is tested against NumPy's, it might create a cycle of testing dependency.

Reply:
> Do you want to do this in the same PR?

No, just a suggestion if you have the time and inclination to look into it. Writing comprehensive tests is hard work.